rclrs/src/wait.rs

```diff
@@ -126,6 +144,56 @@ impl WaitSet {
         Ok(())
     }
 
+    /// Adds a client to the wait set.
+    ///
+    /// It is possible, but not useful, to add the same client twice.
+    ///
+    /// This will return an error if the number of clients in the wait set is larger than the
+    /// capacity set in [`WaitSet::new`].
+    ///
+    /// The same client must not be added to multiple wait sets, because that would make it
+    /// unsafe to simultaneously wait on those wait sets.
+    pub fn add_client(&mut self, client: Arc<dyn ClientBase>) -> Result<(), RclrsError> {
+        unsafe {
+            // SAFETY: I'm not sure if it's required, but the subscription pointer will remain valid
+            // for as long as the wait set exists, because it's stored in self.subscriptions.
```

Comment:
```suggestion
            // SAFETY: I'm not sure if it's required, but the client pointer will remain valid
            // for as long as the wait set exists, because it's stored in self.clients.
```